plenum/test/primary_selection/test_primary_selector.py

```diff
@@ -334,3 +336,83 @@ def test_send_view_change_done_message(tmpdir):
     print(messages)
 
     assert list(node.view_changer.outBox) == messages
+
+
+nodeCount = 7
+
+
+@pytest.fixture(scope="function")
+def txnPoolNodeSetWithElector(node_config_helper_class,
+                              patchPluginManager,
+                              tdirWithPoolTxns,
+                              tdirWithDomainTxns,
+                              tdir,
+                              tconf,
+                              poolTxnNodeNames,
+                              allPluginsPath,
+                              tdirWithNodeKeepInited,
+                              testNodeClass,
+                              do_post_node_creation):
+    with ExitStack() as exitStack:
+        nodes = []
+        for nm in poolTxnNodeNames:
+            node = exitStack.enter_context(create_new_test_node(
+                testNodeClass, node_config_helper_class, nm, tconf, tdir,
+                allPluginsPath))
+            do_post_node_creation(node)
+            nodes.append(node)
+        for node in nodes:
+            node.view_changer = node.newViewChanger()
+            node.elector = node.newPrimaryDecider()
+        yield nodes
+
+
+def test_primaties_selection(txnPoolNodeSetWithElector):
```

Comment:
Please split the test into a multiple of small tests.

Reply:
Done.